cpp/src/text/edit_distance.cu

```diff
@@ -101,34 +100,27 @@ struct edit_distance_levenshtein_algorithm {
     auto d_str =
       d_strings.is_null(idx) ? cudf::string_view{} : d_strings.element<cudf::string_view>(idx);
     auto d_tgt = [&] __device__ {  // d_targets is also allowed to have only one entry
-      if (d_targets.is_null(idx)) { return cudf::string_view{}; }
+      if (d_targets.size() > 1 && d_targets.is_null(idx)) { return cudf::string_view{}; }
       return d_targets.size() == 1 ? d_targets.element<cudf::string_view>(0)
                                    : d_targets.element<cudf::string_view>(idx);
     }();
     d_results[idx] = compute_distance(d_str, d_tgt, d_buffer + d_offsets[idx]);
   }
 };
 
-struct edit_distance_matrix_levenshtein_algorithm {
-  cudf::column_device_view d_strings;  // computing these against itself
-  cudf::size_type* d_buffer;           // compute buffer for each string
-  std::ptrdiff_t const* d_offsets;     // locate sub-buffer for each string
-  cudf::size_type* d_results;          // edit distance values
+struct calculate_compute_buffer_fn {
+  cudf::column_device_view d_strings;
+  cudf::column_device_view d_targets;
 
-  __device__ void operator()(cudf::size_type idx) const
+  __device__ std::ptrdiff_t operator()(cudf::size_type idx) const
```

Comment:
`cuda::std::` instead of `std::`?

Reply:
No. This is a type like `std::size_t` and not a function.